contracts/Archetype.sol

```diff
@@ -112,24 +112,28 @@ contract Archetype is
     address affiliate,
     bytes calldata signature
   ) external payable {
-    if (quantityList.length != toList.length) {
-      revert InvalidConfig();
-    }
+    require(quantityList.length == toList.length, "InvalidConfig");
 
     DutchInvite storage invite = invites[auth.key];
-    uint256 quantity = 0;
-    {
-      uint32 unitSize = invite.unitSize;
-      for (uint256 i = 0; i < quantityList.length; i++) {
-        if (unitSize > 1) {
-          quantity += quantityList[i] * unitSize;
-        } else {
-          quantity += quantityList[i];
-        }
+    uint256 curSupply = _totalMinted();
+    uint256 quantity;
+
+    for (uint256 i; i < toList.length; ) {
+      uint256 quantityToAdd;
+      if (invite.unitSize > 1) {
+        quantityToAdd = quantityList[i] * invite.unitSize;
+      } else {
+        quantityToAdd = quantityList[i];
+      }
+      quantity += quantityToAdd;
+
+      _mint(toList[i], quantityToAdd);
+
+      unchecked {
+        ++i;
       }
```

Comment:
This saves some gas? Any downsides?

Reply:
Nope this is totally safe to do since you will never overflow due to you never minting over the max uint256 number of addresses